ContactControles.js

```diff
@@ -0,0 +1,142 @@
+var app = angular.module('myApp', ["ngRoute"]);
+
+app.config(function($routeProvider,$locationProvider) {
+     $routeProvider.when('/', {
+               templateUrl : 'home.html',
+               controller: 'contactsCtrl'
+     }).when('/editContact/:contact_index', {
+               templateUrl : 'addContact.html',
+               controller: 'editCtrl'
+     }).when('/addContact', {
+               templateUrl : 'addContact.html',
+               controller: 'addCtrl'
+                         }).otherwise({
+         redirectTo: '/'
+                         });
+ });
+
+app.controller('contactsCtrl', function($scope, ContactService, $http, $location, $window) {

```

Comment:
- Naming convention: since this controller is responsible for listing all contacts, I believe it's name should be changed t something list `contactsListController`, always keep in mind to assign meaningful names in your declaration, it will help you easily remember what this controller / function does.
- I refactored this controller to give you an indentation example.